tests/test_direct_chat_async.py

```
@@ -38,6 +38,18 @@ def test_agent_mode_queues_async_job(monkeypatch, tmp_path: Path):
     monkeypatch.setattr(direct_chat, "_agent_jobs", AgentJobManager())
     monkeypatch.setattr(direct_chat, "_agent_workspace_root", tmp_path / "workspaces")
 
+    # Stub PROVIDER_ROUTER so _run_agent_job can resolve provider settings
+    class _FakeProvider:
+        priority = 1
+        api_key = None
+        normalized_base_url = "http://localhost:11434"
+        def auth_headers(self): return {}
+
+    class _FakeRouter:
+        providers = [_FakeProvider()]
```

Ruff (0.15.12)

[warning] Mutable default value for class attribute

(RUF012)